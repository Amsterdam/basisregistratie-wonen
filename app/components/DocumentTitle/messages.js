import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  default_title: {
    id: `${scope}.registration_plural`,
  },
});
