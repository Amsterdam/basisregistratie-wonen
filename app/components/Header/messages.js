/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  logged_in_as: {
    id: `${scope}.logged_in_as`,
  },
  log_in: {
    id: `${scope}.log_in`,
  },
  log_in_adw: {
    id: `${scope}.log_in_adw`,
  },
  log_out: {
    id: `${scope}.log_out`,
  },
});
