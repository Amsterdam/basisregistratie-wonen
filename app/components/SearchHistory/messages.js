/*
 * SearchHistory Messages
 *
 * This contains all the text for the SearchHistory component.
 */
import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  search_history_header: {
    id: `${scope}.search_history_header`,
    defaultMessage: 'Recently search for',
  },
});
