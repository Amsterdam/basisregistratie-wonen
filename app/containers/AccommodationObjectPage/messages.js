/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  yes: {
    id: 'boilerplate.containers.AccommodationObjectPage.yes',
    defaultMessage: 'Ja',
  },
  no: {
    id: 'boilerplate.containers.AccommodationObjectPage.no',
    defaultMessage: 'Nee',
  },
  does_not_apply: {
    id: 'boilerplate.containers.AccommodationObjectPage.does_not_apply',
    defaultMessage: 'n.v.t.',
  },
});
