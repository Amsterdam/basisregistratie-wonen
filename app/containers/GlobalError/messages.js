/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  unable_to_fetch: {
    id: `${scope}.unable_to_fetch`,
  },
  unauthorized: {
    id: `${scope}.unauthorized`,
  },
  session_expired: {
    id: `${scope}.session_expired`,
  },
});
