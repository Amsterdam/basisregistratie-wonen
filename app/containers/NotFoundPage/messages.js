/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  document_title: {
    id: `${scope}.registration_plural`,
  },
  header: {
    id: `${scope}.page_not_found`,
  },
});
