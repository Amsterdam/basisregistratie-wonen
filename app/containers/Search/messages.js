import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  search_placeholder: {
    id: `${scope}.search_placeholder`,
  },
});
