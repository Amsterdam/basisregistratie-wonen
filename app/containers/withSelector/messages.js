/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  yes: {
    id: 'boilerplate.containers.AccommodationObjectPage.yes',
    defaultMessage: 'Ja',
  },
  no: {
    id: 'boilerplate.containers.AccommodationObjectPage.no',
    defaultMessage: 'Nee',
  },
  unknown: {
    id: 'boilerplate.containers.AccommodationObjectPage.unknown',
    defaultMessage: 'n.b.',
  },
});
