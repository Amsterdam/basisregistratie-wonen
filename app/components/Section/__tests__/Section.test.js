import React from 'react';
import { render, cleanup } from 'react-testing-library';
import 'jest-styled-components';
import { IntlProvider } from 'react-intl';
import { isArray } from 'utils';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import configureStore from '../../../configureStore';
import { SectionComponent as Section } from '..';
import messages from '../../../translations/nl.json';
import pand from './pand.json';
import subjectNP from './subjectNP.json';

const store = configureStore({}, browserHistory);
const intlProvider = new IntlProvider({ locale: 'nl', messages });
const { intl } = intlProvider.getChildContext();

jest.mock('utils', () => {
  const actual = require.requireActual('utils');

  return {
    ...actual,
    isArray: jest.fn(),
  };
});

describe('Section', () => {
  afterEach(cleanup);

  it('matches the snapshot', () => {
    const { container } = render(<Section name="Qux baz" intl={intl} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render an external link', () => {
    const { container, rerender } = render(<Section name="Qux baz" intl={intl} />);

    const anchors = container.getElementsByTagName('a');
    expect(anchors).toHaveLength(0);

    rerender(<Section name="Qux baz" href="https://www.amsterdam.nl" intl={intl} />);

    expect(anchors).toHaveLength(1);
    expect(anchors[0].getAttribute('target')).toBe('_blank');
    expect(anchors[0].getAttribute('rel')).toBe('noopener noreferrer');
  });

  it('should render a loading indicator', () => {
    const { rerender, queryByTestId } = render(<Section name="Bar bar baz" intl={intl} />);

    expect(queryByTestId('progress-wrapper')).not.toBeNull();

    rerender(<Section name="Bar bar baz" intl={intl} data={null} />);

    expect(queryByTestId('progress-wrapper')).toBeNull();
  });

  it('should not render the title when there is no data', () => {
    // selectors return 'undefined' when the app saga didn't retrieve the data yet. After retrieval, when the response is empty,
    // the saga returns 'null'
    const { rerender, queryByText } = render(<Section name="Foo bar baz" intl={intl} />);

    expect(queryByText('Foo bar baz')).not.toBeNull();

    rerender(<Section name="Foo bar baz" intl={intl} data={null} />);

    expect(queryByText('Foo bar baz')).toBeNull();
  });

  it('should verify if data items are arrays', () => {
    const data = [pand[0], pand[1]];
    const { rerender } = render(
      <Provider store={store}>
        <IntlProvider locale="nl" messages={messages}>
          <Section name="Foo bar baz" intl={intl} data={data} />
        </IntlProvider>
      </Provider>,
    );

    expect(isArray).toHaveBeenCalledTimes(4); // will call isArrayOfArrays
    expect(isArray).toHaveBeenCalledWith(data[0]);
    expect(isArray).toHaveBeenLastCalledWith(data[1]);
    expect(document.getElementsByTagName('li')).toHaveLength(2);

    isArray.mockReset();

    const data2 = [[pand[2]]];

    rerender(
      <Provider store={store}>
        <IntlProvider locale="nl" messages={messages}>
          <Section name="Foo bar baz" intl={intl} data={data2} />
        </IntlProvider>
      </Provider>,
    );

    expect(isArray).toHaveBeenCalledTimes(3);
    expect(isArray).toHaveBeenCalledWith(data2[0]);
    expect(document.getElementsByTagName('li')).toHaveLength(1);

    isArray.mockReset();

    const data3 = subjectNP;

    render(
      <Provider store={store}>
        <IntlProvider locale="nl" messages={messages}>
          <Section name="Foo bar baz" intl={intl} data={data3} />
        </IntlProvider>
      </Provider>,
    );

    expect(isArray).toHaveBeenCalledTimes(4);
    expect(document.getElementsByTagName('ul')).toHaveLength(2);
  });

  it('should render section contents', () => {
    const data = pand;
    const numValues = data.length;

    render(
      <Provider store={store}>
        <IntlProvider locale="nl" messages={messages}>
          <Section name="Foo bar baz" intl={intl} data={data} />
        </IntlProvider>
      </Provider>,
    );

    expect(document.getElementsByTagName('li')).toHaveLength(numValues);
  });
});
