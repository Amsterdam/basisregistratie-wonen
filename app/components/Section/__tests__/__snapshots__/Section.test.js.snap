// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section matches the snapshot 1`] = `
.c2 {
  position: relative;
}

.c2.hidden {
  display: none;
}

.c3 {
  height: 19.98px;
  width: 19.98px;
  border-radius: 50%;
  box-shadow: inset 0 0 0 1.9980000000000002px;
  color: #ec0000;
  display: inline-block;
  font-size: 1.9980000000000002px;
  position: relative;
  text-indent: -99999em;
  -webkit-transform: translate3D(0,0,0);
  -ms-transform: translate3D(0,0,0);
  transform: translate3D(0,0,0);
  vertical-align: middle;
  margin: 0;
}

.c3::before,
.c3::after {
  background-color: #fff;
  content: '';
  height: 21.978px;
  position: absolute;
}

.c3::before {
  -webkit-animation: rotate 1.5s infinite ease 1.2s;
  animation: rotate 1.5s infinite ease 1.2s;
  border-radius: 21.978px 0 0 21.978px;
  left: -0.9990000000000001px;
  top: -0.9990000000000001px;
  -webkit-transform-origin: 10.589400000000001px 10.989px;
  -ms-transform-origin: 10.589400000000001px 10.989px;
  transform-origin: 10.589400000000001px 10.989px;
  width: 10.3854375px;
}

.c3::after {
  -webkit-animation: rotate 1.5s infinite ease;
  animation: rotate 1.5s infinite ease;
  border-radius: 0 21.978px 21.978px 0;
  left: 10.3854375px;
  top: -0.5994px;
  -webkit-transform-origin: -0.1998px 10.989px;
  -ms-transform-origin: -0.1998px 10.989px;
  transform-origin: -0.1998px 10.989px;
  width: 11.186735400000002px;
}

.c1 {
  margin-top: 30px;
  margin-top: 0;
}

.c0 {
  position: relative;
}

@media print {
  .c1 {
    margin-top: 0;
  }
}

<section
  class="c0"
>
  <h3
    class="c1"
    id="Qux baz"
  >
    Qux baz
    
  </h3>
  <div
    class="no-print  c2"
    data-testid="progress-wrapper"
    type="undetermined"
  >
    <div
      class="c3"
    />
  </div>
</section>
`;

exports[`Section should render section contents 1`] = `
.c1 {
  margin-top: 30px;
  margin-top: 0;
}

.c4 {
  font-weight: bold;
  margin-top: 20px;
}

.c3 {
  padding: 0;
  list-style: none;
  padding-bottom: 30px;
}

.c3:not(:first-child) {
  border-bottom: 4px solid #767676;
}

.c3 li:last-of-type ul {
  padding-bottom: 0;
}

.c3 *:last-of-type > *:last-of-type ul:last-of-type {
  border: 0;
}

.c3 li {
  padding-left: 23px;
  position: relative;
}

.c3 li:before {
  content: '';
  position: absolute;
  left: 0;
  width: 8px;
  height: 8px;
  margin-top: 7px;
  background-color: #767676;
}

.c3 .c2 li {
  padding-left: 0;
}

.c3 .c2 li:before {
  content: unset;
}

.c0 {
  position: relative;
}

@media print {
  .c1 {
    margin-top: 0;
  }
}

<section
  class="c0"
>
  <h3
    class="c1"
    id="Foo bar baz"
  >
    Foo bar baz
    
  </h3>
  <ul
    class="c2 c3"
  >
    <li>
      <strong
        class="c4"
        lang="nl"
      >
        Pandidentificatie
      </strong>
      : 
      0363100012096240
    </li>
    <li>
      <strong
        class="c4"
        lang="nl"
      >
        Status
      </strong>
      : 
      Pand in gebruik
    </li>
    <li>
      <strong
        class="c4"
        lang="nl"
      >
        Oorspronkelijk bouwjaar
      </strong>
      : 
      1921
    </li>
    <li>
      <strong
        class="c4"
        lang="nl"
      >
        Hoogste bouwlaag
      </strong>
      : 
      n.b.
    </li>
    <li>
      <strong
        class="c4"
        lang="nl"
      >
        Laagste bouwlaag
      </strong>
      : 
      n.b.
    </li>
    <li>
      <strong
        class="c4"
        lang="nl"
      >
        Aantal verblijfsobjecten
      </strong>
      : 
      <span>
        2
      </span>
    </li>
  </ul>
</section>
`;
